### ISSUE TITLE
getting <unk> for all words

### ISSUE THREAD

**Issue by sriram-c** [tensorflow/nmt#62]
Hi,

I am training on English-Hindi pair of around 1.5 million parallel sent. The training was done without giving any error. 

But while translating test sent, I am getting all <unk> <unk> .

I have taken a very small vocabulary file for Hindi . i.e. size of 8k words.

Is this the reason? 

Thanks,
Sriram

**Comment by sriram-c** [comment]
sorry,
the above sentence was not complete.
But while translating test sent, I am getting all "<unk>"

**Comment by sriram-c** [comment]
I have succesfully trained the Vietnamese-English system as given in the tutorial.

**Comment by sriram-c** [comment]
I have increased the vocabulary size (sort and uniq of all the words in train file).

The \<unk\> has disappeared to some extent. Still the bleu score is quite low for test set (0.1).

what may be the possible reason?

**Comment by ssokhey** [comment]
How are you building your vocabulary? 

You should generate vocabulary from training data. You will get much larger vocabulary (without subword) for 1.5 million parallel sentences (I hope you are using IITB parallel corpus).

**Comment by Sabyasachi18** [comment]
@ssokhey ....I have a question. I want to run incremental training on my trained German-English Engine.
Can I update my vocab file with new words from the incremental training data. If Yes, then kindly let me know the process. 

Should I append the new words at the end of the existing vaocbulary file?
Or should i  do a sorting of the vocab file after appending the new words to it?
